bpf/lib/tunnel.h

```diff
@@ -83,4 +88,135 @@ tunnel_vni_to_sec_identity(__be32 vni)
 	return bpf_ntohl(vni) >> 8;
 }
 
+/*
+ * Returns true if the skb associated with data pointers is a vxlan encapsulated
+ * packet.
+ *
+ * The determination is made by comparing the UDP destination port with
+ * the tunnel_port provided to the function.
+ */
+static __always_inline bool
+tunnel_skb_is_vxlan_v4(const void *data, const void *data_end,
+		       const struct iphdr *ipv4, const __u16 tunnel_port)
+{
+	struct udphdr *udp = NULL;
+	__u32 l3_size = 0;
+
+	if (ipv4->protocol != IPPROTO_UDP)
+		return false;
+
+	l3_size = ipv4->ihl * 4;
```

Comment:
```suggestion
	l3_size = ipv4_hdrlen(ipv4);
```

(please check for other occurrences)

Reply:
Ugh, this issue I face here is including `#include "ipv4.h"` 
Then includes `dbg.h` and `map.h` and others and without following more include chains the compilation fails:

```
In file included from /home/louis/git/gopath/src/github.com/cilium/cilium-enterprise/bpf/lib/ipv4.h:10:
/home/louis/git/gopath/src/github.com/cilium/cilium-enterprise/bpf/lib/l4.h:60:10: error: use of undeclared identifier 'DROP_CSUM_L4'
   60 |                 return DROP_CSUM_L4;
      |                        ^
/home/louis/git/gopath/src/github.com/cilium/cilium-enterprise/bpf/lib/l4.h:63:10: error: use of undeclared identifier 'DROP_WRITE_ERROR'
   63 |                 return DROP_WRITE_ERROR;
      |                        ^
In file included from bpf_lxc.c:20:
In file included from ./lib/auth.h:7:
In file included from ./lib/common.h:21:
In file included from ./lib/tunnel.h:13:
In file included from /home/louis/git/gopath/src/github.com/cilium/cilium-enterprise/bpf/lib/ipv4.h:11:
In file included from /home/louis/git/gopath/src/github.com/cilium/cilium-enterprise/bpf/lib/metrics.h:13:
In file included from /home/louis/git/gopath/src/github.com/cilium/cilium-enterprise/bpf/lib/maps.h:8:
/home/louis/git/gopath/src/github.com/cilium/cilium-enterprise/bpf/lib/ipv6.h:54:11: error: use of undeclared identifier 'DROP_INVALID_EXTHDR'
   54 |                         return DROP_INVALID_EXTHDR;
      |                                ^
/home/louis/git/gopath/src/github.com/cilium/cilium-enterprise/bpf/lib/ipv6.h:57:11: error: use of undeclared identifier 'DROP_FRAG_NOSUPPORT'
   57 |                         return DROP_FRAG_NOSUPPORT;
      |                                ^
/home/louis/git/gopath/src/github.com/cilium/cilium-enterprise/bpf/lib/ipv6.h:64:12: error: use of undeclared identifier 'DROP_INVALID'
   64 |                                 return DROP_INVALID;
      |                                        ^
/home/louis/git/gopath/src/github.com/cilium/cilium-enterprise/bpf/lib/ipv6.h:81:9: error: use of undeclared identifier 'DROP_INVALID_EXTHDR'
   81 |         return DROP_INVALID_EXTHDR;
      |                ^
  ```
  
  If we want to have helper functions like this, they should really be within 'export only' headers (or at least headers which include other minimal headers), we have created a big nest of include declarations which I don't want to fix in this PR.

Reply:
Going to resolve this for now. I have a personal endeavor to make this situation better, once I unravel this stuff in a separate PR I will replace suggested line above (the suggestion is good).